### PR TITLE
add api to reconfigure buffer duration based on storage size

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/StreamInfoProvider.c [hunk 1]
```diff
@@ -216,3 +216,17 @@ STATUS createAudioVideoStreamInfo(STREAMING_TYPE streamingType, PCHAR streamName
     return retStatus;
 }
 
+STATUS configStreamInfoBasedOnStorageSize(UINT32 storageSize, UINT32 avgBitrate, UINT32 totalStreamCount, PStreamInfo pStreamInfo)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CHK(pStreamInfo != NULL, STATUS_NULL_ARG);
+
+    pStreamInfo->streamCaps.bufferDuration = (UINT64) ((DOUBLE) storageSize / 1024 * 8 / avgBitrate / totalStreamCount * 0.85) * HUNDREDS_OF_NANOS_IN_A_SECOND;
```
Comment (MushMal) [inline src/source/StreamInfoProvider.c]: Use a constant for 0.85.

File: samples/KvsVideoOnlyStreamingSample.c [hunk 2]
```diff
@@ -91,8 +93,10 @@ INT32 main(INT32 argc, CHAR *argv[])
     CHK_STATUS(createDefaultDeviceInfo(&pDeviceInfo));
     // adjust members of pDeviceInfo here if needed
     pDeviceInfo->clientInfo.loggerLogLevel = LOG_LEVEL_DEBUG;
+    pDeviceInfo->storageInfo.storageSize = DEFAULT_STORAGE_SIZE;
```
Comment (disa6302) [inline samples/KvsVideoOnlyStreamingSample.c]: Is setting this necessary. Is it not possible to do the calculation with the 128MB default?

File: CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ project(KinesisVideoProducerC LANGUAGES C)
 # User Flags
 option(ADD_MUCLIBC "Add -muclibc c flag" OFF)
 option(BUILD_DEPENDENCIES "Whether or not to build depending libraries from source" ON)
-option(BUILD_OPENSSL "If building dependencies, whether or not building openssl from source" OFF)
+option(BUILD_OPENSSL "If building dependencies, whether or not building openssl from source" ON)
```
Comment (disa6302) [inline CMakeLists.txt]: Is this change related to what the current patch intends to do?

File: samples/KvsVideoOnlyStreamingSample.c [hunk 4]
```diff
@@ -6,6 +6,8 @@
 #define DEFAULT_KEY_FRAME_INTERVAL          45
 #define DEFAULT_FPS_VALUE                   25
 #define DEFAULT_STREAM_DURATION             20 * HUNDREDS_OF_NANOS_IN_A_SECOND
+#define DEFAULT_STORAGE_SIZE                20 * 1024 * 1024
+#define RECORDED_FRAME_AVG_BITRATE_KBIT_PS  3800
```
[the remark this answers is not in the document]
Comment (chehefen) [reply samples/KvsVideoOnlyStreamingSample.c]: yes the sample bitrate is about 3.8mbps. changed to bps

File: src/include/com/amazonaws/kinesis/video/cproducer/Include.h [hunk 5]
```diff
@@ -458,6 +458,18 @@ PUBLIC_API STATUS createRealtimeAudioVideoStreamInfoProvider(PCHAR, UINT64, UINT
...
+ * Configure streaminfo based on given storage size and average bitrate
+ * Will change buffer duration, stream latency duration.
+ *
+ * @param - UINT32 - Storage size in bytes
+ * @param - UINT32 - Total average bitrate for all tracks. Unit: kb/s
+ * @param - UINT32 - Total number of streams per kinesisVideoStreamClient
+ * @param - PStreamInfo
+ * @return - STATUS code of the execution
+*/
+PUBLIC_API STATUS configStreamInfoBasedOnStorageSize(UINT32, UINT32, UINT32, PStreamInfo);
```
[the remark this answers is not in the document]
Comment (chehefen) [reply src/include/com/amazonaws/kinesis/video/cproducer/Include.h]: that would break backward compatibility.

[the remark this answers is not in the document]
Comment (chehefen) [reply src/include/com/amazonaws/kinesis/video/cproducer/Include.h]: done

File: samples/KvsVideoOnlyStreamingSample.c (hunk 2 above)
Comment (chehefen) [reply samples/KvsVideoOnlyStreamingSample.c]: I think people reading the sample wouldnt know about the 128MB default. Also this way they can change the storage size and the buffer duration would still be right.

File: CMakeLists.txt (hunk 3 above)
Comment (chehefen) [reply CMakeLists.txt]: removed.
